systemtable.go

```diff
@@ -6,46 +6,70 @@ import (
 	"github.com/rivo/tview"
 	"net/http"
 	"strconv"
+	"sync"
 	"time"
 )
 
 const (
-	statusUp   = "[green]UP"
-	statusDown = "[red]DOWN"
-	statusErr  = "[red]ERR"
-	statusUnk  = "[gray]UNK"
+	statusUp = "[green]UP"
 
 	nameCol      = 0
 	statusCol    = 1
 	intervalCol  = 2
 	failCountCol = 3
 )
 
-func DisplayTable(app *tview.Application) *tview.Table {
+type systemTable struct {
+	*tview.Table
+
+	app   *tview.Application
+	pages *tview.Pages
+
+	mux     sync.Mutex
+	systems []*System
+}
+
+func (t *systemTable) Add(sys System) {
+	rows := t.GetRowCount()
+	t.mux.Lock()
+	t.systems = append(t.systems, &sys)
+	t.mux.Unlock()
+	t.Table.SetCell(rows, nameCol, tview.NewTableCell(sys.Name))
+	t.Table.SetCell(rows, statusCol, tview.NewTableCell(Unknown.UiString()))
+	t.Table.SetCell(rows, intervalCol, tview.NewTableCell(sys.CheckInterval))
+	t.Table.SetCell(rows, failCountCol, tview.NewTableCell(strconv.Itoa(len(sys.FailedChecks))).SetAlign(tview.AlignRight))
+
+	switch {
+	case sys.Http != nil:
+		go httpPollingLoop(t.app, t.Table, &sys, rows)
+	}
+}
+
+func DisplayTable(app *tview.Application, pages *tview.Pages, systems ...System) *systemTable {
```

Comment:
Rename to `NewSystemTable` to make it more clear that this isn't doing the displaying.